WebbPSF to STPSF

BradleySappington opened the pull request:
transition webbpsf to stpsf alias

BradleySappington committed:
import stpsf

pep8speaks commented:
Hello @BradleySappington, Thank you for updating !



There are no PEP8 issues in this Pull Request.

##### Comment last updated at 2024-12-19 15:30:54 UTC

BradleySappington committed:
update workflow

BradleySappington committed:
read the docs redirect

mperrin commented:
Read the docs redirect. Is there a way to make that global across all pages? Would we need to put that redirect text on each of the individual RST pages? 

Thanks so much for all the work on this -- so many little details to keep track of!

BradleySappington committed:
remove webbpsf conftest.py

BradleySappington commented:
@mperrin - I don't actually think this readthedocs redirect is needed.  I created a domain redirect through readthedocs admin settings and have successfully tested (I just have it deactivated currently)

mperrin commented:
superb, I'm glad to hear that there's a way to do a domain redirect, and that you've already figured it out! :-)

BradleySappington committed:
update workflows

BradleySappington committed:
simplify tox

BradleySappington commented:
I removed some of the workflows as we are now importing everything from stpsf (including tests) they are failing here.  No sense in spending time making the workflows functional to archive the repo in a few months so I removed them.  Will not merge until confirmed everyone is comfortable with these changes.

mperrin commented on docs/index.rst:
```diff
@@ -1,113 +1,2 @@
-
-If you would like to receive email announcements of future versions, please contact Marshall Perrin, or visit `maillist.stsci.edu <https://maillist.stsci.edu/scripts/wa.exe?SUBED1=webbpsf-users&A=1>` to subscribe yourself to the "webbpsf-users@maillist.stsci.edu" list.
+.. meta::
+   :http-equiv=Refresh: 0; url='https://stpsf.readthedocs.io/en/latest/'
```
Did you decide this redirect was needed, or were you going to use the domain redirect option in readthedocs that you mentioned?  Either way is fine with me - I just want to make sure we all understand what the plan is and how it's going to work.

Relatedly, I see the docs build was left in the CI tests. That makes me think that the plan is _not_ to use a domain redirect in readthedocs, since if we're just redirecting the whole domain, why do we need to test that these docs build?

BradleySappington replied:
I'd like to stick with the redirect through readthedocs.  I had made this change initially before realizing I could do it on read the docs.  I chose not to change it back because I liked the code showing the intention of not using this documentation any more, however did not feel it necessary to delete the entire documentation base.  If you'd like me to put back please advise.

BradleySappington committed:
remove workflows

york-stsci reviewed (approved):
Seems like a reasonable way to start the sunset process

mperrin commented on tox.ini:
```diff
@@ -1,42 +1,8 @@
-    py{310,311,312}-{poppydev,pysiafdev,astropydev,latest,stable}-test
-    py310-legacy-test
-    py{310,311,312}-{poppydev,pysiafdev}-cov
+envlist = docbuild
```
This doesn't seem right to me to remove all test environments apart from the docs. 

We're still going to have the `webbpsf` package as a wrapper that imports `stpsf`, and being able to do `import webbpsf` and have existing code run is a KEY PART of the intended functionality here. 

Therefore there needs to be a test suite that verifies that we can indeed `import webbpsf` and run calculations. 

So please let's keep at least one test environment that runs the existing full test suite of webbpsf test functions.  I don't think this needs to have all the different instances in the test matrix -- those will be covered by the stpsf test suite -- but there should be enough here to test that the import wrapper is working as intended.

BradleySappington replied:
Context for removing all the test environments:

Because the entire repository is replaced with stpsf, the tests that are running are stpsf tests.  This wasn't playing nicely with our CI as all tests labelled webbpsf no longer existed.  As this was intended to be the final release of WebbPSF, I decided to manually verify that the init change worked properly (as these automated tests will not need to run in the future because development is stopping in this repository).  I think that if we all take a look and are comfortable with testing the build manually, that suffices.  If this makes anyone uncomfortable I can spend time trying to get the CI functional, however I felt the benefits of that time investment were minimal.

BradleySappington committed:
update warning

BradleySappington committed:
update warning

BradleySappington committed:
only warn once

BradleySappington committed:
spelling

BradleySappington committed:
Simplify warning

mperrin commented on webbpsf/__init__.py:
```diff
@@ -13,119 +13,42 @@
-from .roman import WFI, RomanCoronagraph  # noqa
-
-from .match_data import setup_sim_to_match_file  # noqa
+os.environ["STPSF_PATH"] = os.environ.get("WEBBPSF_PATH", "")
```
Oh this is very clever!  I had thought we would need code in `stpsf` to check WEBBPSF_PATH and if needed use that for STPSF_PATH. But doing it here as part of this import wrapper takes care of that so succinctly. I like this a lot.

BradleySappington replied:
thanks!

mperrin commented on webbpsf/__init__.py:
```diff
@@ -13,119 +13,42 @@
+_warned = False
 
+if not _warned:
+    warnings.warn(
```
As discussed in slack, I'm not sure a multi-line warning with lots of `!!!!!` is the best choice at this time.  Suggest making it shorter, and making a webpage in the docs for more information on the name migration. 

Perhaps like: 
```
"""
This message is for information only and WebbPSF will continue to function as normal.
The WebbPSF library has been moved/renamed to STPSF. Please see [READTHEDOCS_PAGE_TBD] for more information. 
WebbPSF is now an alias of STPSF and is running code from the STPSF library.
"""

```

And then that webpage, some new child page within the stpsf docs, can give more details and example code, rather than trying to pack too much into a warning message.

BradleySappington replied:
Replacing exactly as suggested here with a TODO until we have a proper page with instructions set up (stpsf 2.0.1)

mperrin reviewed (changes requested):
Good overall, but (a) I think we should not go quite so far in cutting out the test suite, and (b) as discussed in slack let's make a docs page about the name migration rather than trying to put all the info in a warning message.

BradleySappington committed:
update warning

BradleySappington committed:
Update warning message to include link

BradleySappington committed:
re-establish testing suite

BradleySappington committed:
update deprecated action

BradleySappington committed:
remove coverage

BradleySappington commented:
Some tests marked expected and not actually running, trying a reboot

BradleySappington committed:
add tests back to ci_workflows

BradleySappington committed:
try to not mess with tests

BradleySappington committed:
temp check for tox update

BradleySappington committed:
update __init__ with proper link

BradleySappington committed:
update test suite

BradleySappington committed:
update ci_workflow env

BradleySappington committed:
update test

BradleySappington committed:
add import-mode

BradleySappington committed:
another try to get local test running

BradleySappington committed:
another take with a setenv

BradleySappington committed:
and agian

BradleySappington committed:
remove tests

BradleySappington commented:
@mperrin - At your convenience, can you please update requested change as complete per slack conversation. Thanks!

BradleySappington commented:
@obi-wan76 - the requested changes have been addressed via slack.  The tests are passing. I'm going to override the merge restriction as the data-cache update fixed in this PR is currently affecting romancal
